--- ./client-admin/src/components/conversation-admin/stats/index.js
+++ ../patches/client-admin/src/components/conversation-admin/stats/index.js
@@ -12,10 +12,12 @@
 import ComponentHelpers from '../../../util/component-helpers'
 import NoPermission from '../no-permission'
 
+import { withTranslation } from 'react-i18next';
+
 @connect((state) => state.stats)
 @connect((state) => state.zid_metadata)
 @connect((state) => state.stats)
-class ConversationStats extends React.Component {
+class ConversationStatsTranslated extends React.Component {
   constructor(props) {
     super(props)
     const times = dateSetupUtil()
@@ -91,7 +93,9 @@
       !conversation_stats.firstCommentTimes ||
       !conversation_stats.firstVoteTimes
 
-    if (loading) return <Box>Loading...</Box>
+    const { t } = this.props;
+    
+    if (loading) return <Box>{t('stats.loading')}</Box>
 
     return (
       <div>
@@ -102,7 +106,7 @@
             lineHeight: 'body',
             mb: [3, null, 4]
           }}>
-          Monitor
+          {t('stats.title')}
         </Heading>
         <NumberCards data={conversation_stats} />
         <Voters
@@ -120,4 +124,5 @@
   }
 }
 
+const ConversationStats = withTranslation()(ConversationStatsTranslated);
 export default ConversationStats
