--- ./server/src/server.ts
+++ ../patches/server/src/server.ts
@@ -2921,15 +2921,15 @@
 
     // These fields only exist on the PolisWebServer deployment.
     if (Config.applicationName === "PolisWebServer") {
-      let x_forwarded_for = req?.headers?.["x-forwarded-for"];
+      //let x_forwarded_for = req?.headers?.["x-forwarded-for"]; // DigiFinland customization - comment out saving encrypted IP-addresses
       let ip: string | null = null;
-      if (x_forwarded_for) {
-        let ips = x_forwarded_for;
-        ips = ips && ips.split(", ");
-        ip = ips.length && ips[0];
-        info.encrypted_ip_address = encrypt(ip);
-        info.encrypted_x_forwarded_for = encrypt(x_forwarded_for);
-      }
+//      if (x_forwarded_for) {
+//        let ips = x_forwarded_for;
+//        ips = ips && ips.split(", ");
+//        ip = ips.length && ips[0];
+//        info.encrypted_ip_address = encrypt(ip);
+//        info.encrypted_x_forwarded_for = encrypt(x_forwarded_for);
+//      }
     }
 
     if (permanent_cookie) {
