--- ./server/src/server.ts	2023-09-27 12:24:31.430266554 +0300
+++ ../patches/server/src/server.ts	2023-09-27 12:23:42.988023827 +0300
@@ -772,7 +772,7 @@
     },
     next: () => any
   ) {
-    let exempt = devMode;
+    let exempt = true; // DigiFinland customization: enable http-only health checks on GKE, no need to redirect https even on devMode=false since this is done on LB / nginx proxy.
 
     // IE is picky, so use HTTP.
     // TODO figure out IE situation, (proxy static files in worst-case)
@@ -3432,15 +3432,15 @@
     if (referer) {
       info.referrer = referer;
     }
-    let x_forwarded_for = req?.headers?.["x-forwarded-for"];
+    //let x_forwarded_for = req?.headers?.["x-forwarded-for"]; // DigiFinland customization - comment out saving encrypted IP-addresses
     let ip: string | null = null;
-    if (x_forwarded_for) {
-      let ips = x_forwarded_for;
-      ips = ips && ips.split(", ");
-      ip = ips.length && ips[0];
-      info.encrypted_ip_address = encrypt(ip);
-      info.encrypted_x_forwarded_for = encrypt(x_forwarded_for);
-    }
+//    if (x_forwarded_for) {
+//      let ips = x_forwarded_for;
+//      ips = ips && ips.split(", ");
+//     ip = ips.length && ips[0];
+//      info.encrypted_ip_address = encrypt(ip);
+//      info.encrypted_x_forwarded_for = encrypt(x_forwarded_for);
+//    }
     if (permanent_cookie) {
       info.permanent_cookie = permanent_cookie;
     }
@@ -11396,7 +11396,7 @@
       // Cannot find name 'authorUid'. Did you mean 'authoruid'?ts(2552)
       // server.ts(12486, 7): 'authoruid' is declared here.
       // @ts-ignore
-      return "select " + Number(authorUid) + " as uid, 900 as priority";
+      return "select " + Number(authoruid) + " as uid, 900 as priority";
     });
     let authorsQuery: string | null =
       "(" + authorsQueryParts.join(" union ") + ")";
