--- ./client-participation/js/templates/participation.handlebars
+++ ../patches/client-participation/js/templates/participation.handlebars
@@ -187,7 +187,7 @@
          position: relative;
          top: -5px;
          left: 6px;
-        ">closed</span>
+        ">Keskustelu on suljettu</span>
       {{/unless}}
       </h2>
     {{/ifDefined}}
@@ -437,35 +437,9 @@
 {{!  ------------ BEGIN FOOTER ---------------  }}
 {{#unless no_footer}}
 
-  <div style="margin-top: 40px; margin-bottom: 20px;" data-test-footer>
-    <a href="https://pol.is" target="_blank" dir="auto">
-      {{> logo}}
-    </a>
-    <div
-      style="
-        margin-top: 15px;
-        text-align: center;
-      "
-      >
-        {{> linkPrivacy}}
-        {{> linkTos}}
-    </div>
-  </div>
+{{> footer}}
 
-  {{!-- <div style="font-size:18px;font-weight:400;color:white;background-color:#03a9f4;display:flex;-webkit-flex-direction:row;-webkit-flex-wrap:nowrap;-webkit-justify-content:space-between;-webkit-align-items:baseline;-webkit-align-content:stretch;-webkit-order:0;-webkit-flex-grow:0;-webkit-flex-shrink:1;-webkit-flex-basis:auto;-webkit-align-self:auto;" data-reactid=".0.0.$=1$=012.0">
-    <p style="margin-inline-start:20px;font-size:12px;" data-reactid=".0.0.$=1$=012.0.$=1$/=010">© The Authors. 2016 </p>
-    <div style="margin-inline-end:20px;" data-reactid=".0.0.$=1$=012.0.$=1$/=011">
-      <a style="text-decoration:none;cursor:pointer;color:white;font-size:12px;" href="/tos" data-reactid=".0.0.$=1$=012.0.$=1$/=011.$/=10">
-        <span name="file-text-o" class="fa fa-file-text-o" data-reactid=".0.0.$=1$=012.0.$=1$/=011.$/=10.$=10"></span>
-        <span style="margin-inline-start:5px;" data-reactid=".0.0.$=1$=012.0.$=1$/=011.$/=10.$/=11"> TOS </span>
-      </a>
-      <a style="text-decoration:none;cursor:pointer;color:white;font-size:12px;" href="/privacy" data-reactid=".0.0.$=1$=012.0.$=1$/=011.$/=11">
-        <span style="margin-inline-start:5px;" data-reactid=".0.0.$=1$=012.0.$=1$/=011.$/=11.0"> PRIVACY </span>
-      </a>
-    </div>
-  </div> --}}
 {{/unless}}
-  {{! END FOOTER}}
 {{!  ------------ END FOOTER ---------------  }}
 
 </div> {{! END PARTICIPATION VIEW DIV }}
