--- ./client-participation/js/templates/participation.handlebars	2023-04-12 10:45:38.163588000 +0300
+++ ../patches/client-participation/js/templates/participation.handlebars	2023-04-13 10:25:22.256440000 +0300
@@ -187,7 +187,7 @@
          position: relative;
          top: -5px;
          left: 6px;
-        ">closed</span>
+        ">Keskustelu on suljettu</span>
       {{/unless}}
       </h2>
     {{/ifDefined}}
@@ -435,7 +435,8 @@
 
 
 {{!  ------------ BEGIN FOOTER ---------------  }}
-{{#unless no_footer}}
+{{! inverted 'unless' to 'if' for hiding footer here: #unless no_footer }}
+{{#if no_footer}}
 
   <div style="margin-top: 40px; margin-bottom: 20px;" data-test-footer>
     <a href="https://pol.is" target="_blank" dir="auto">
@@ -464,7 +465,7 @@
       </a>
     </div>
   </div> --}}
-{{/unless}}
+{{/if}}
   {{! END FOOTER}}
 {{!  ------------ END FOOTER ---------------  }}
 
