--- ./client-participation/js/templates/vote-view.handlebars
+++ ../patches/client-participation/js/templates/vote-view.handlebars
@@ -55,7 +55,7 @@
       </form>
     {{else}} {{!-- !canSubscribe --}}
       <div><strong>{{txt1}}</strong></div>
-      <div>{{txt2}}</div>
+      <div class="displayNone">{{txt2}}</div>
     {{/if}} {{!-- /canSubscribe --}}
   {{/if}} {{!-- /subscribed --}}
   </div> {{!-- /empty scenario div --}}
