--- ./client-participation/js/strings/en_us.js
+++ ../patches/client-participation/js/strings/en_us.js
@@ -196,6 +196,16 @@
 s.topic_bad_02 = "what are our launch blockers?";
 s.topic_bad_02_reason = "";
 
+s.footer_title = "Polis";
+s.footer_desc = "DigiFinland and Sitra have introduced the Polis online platform in Finland, based on open source code. The purpose of Polis is to enable large groups to participate in constructive exchanges of opinions around selected topics.";
+s.footer_disclaimer = "Polis-platform is in the experimental phase and does not yet meet all the requirements in terms of accessibility and usability.";
+s.footer_version = "Version";
+s.footer_released = "Release date";
+s.footer_provider = "Service provider";
+s.footer_links_privacy = "DigiFinland Oy data protection";
+s.footer_links_info = "Information";
+s.footer_links_source = "Sourcecode";
+
 // Old unused error message from when Polis was a demo <3
 
 s.notSentSinceDemo = "(not really, this is a demo)";
