--- ./client-admin/src/components/conversation-admin/conversation-config.js
+++ ../patches/client-admin/src/components/conversation-admin/conversation-config.js
@@ -156,23 +156,32 @@
           Show explanation text above voting and visualization
         </CheckboxField>
 
-        <CheckboxField
-          field="subscribe_type"
-          label="Prompt participants to subscribe to updates"
-          isIntegerBool>
-          Prompt participants to subscribe to updates. A prompt is shown to
-          users once they finish voting on all available comments. If enabled,
-          participants may optionally provide their email address to receive
-          notifications when there are new comments to vote on.
-        </CheckboxField>
 
-        <CheckboxField field="auth_opt_fb" label="Facebook login prompt">
-          Show Facebook login prompt
-        </CheckboxField>
+        {
+        // hide these settings to avoid accidentally collecting of PII data
+        //
+        // database schema altered to use subscription_type = 0 as default. (not implemented in polis-server constants.ts)
+        // <CheckboxField
+        //   field="subscribe_type"
+        //   label="Prompt participants to subscribe to updates"
+        //   isIntegerBool>
+        //   Prompt participants to subscribe to updates. A prompt is shown to
+        //   users once they finish voting on all available comments. If enabled,
+        //   participants may optionally provide their email address to receive
+        //   notifications when there are new comments to vote on.
+        // </CheckboxField>
+
+        // changed also in polis-server constants.ts: auth_opt_fb = false
+        // <CheckboxField field="auth_opt_fb" label="Facebook login prompt">
+        //   Show Facebook login prompt
+        // </CheckboxField>
+
+        // changed also in polis-server constants.ts: auth_opt_tw = false
+        // <CheckboxField field="auth_opt_tw" label="Twitter login prompt">
+        //   Show Twitter login prompt
+        // </CheckboxField>
+        }
 
-        <CheckboxField field="auth_opt_tw" label="Twitter login prompt">
-          Show Twitter login prompt
-        </CheckboxField>
 
         <Heading
           as="h6"
@@ -188,17 +197,23 @@
           No comments shown without moderator approval
         </CheckboxField>
 
-        <CheckboxField
-          field="auth_needed_to_write"
-          label="Require Auth to Comment">
-          Participants cannot submit comments without first connecting either
-          Facebook or Twitter
-        </CheckboxField>
+        {
+        // hide these settings to avoid accidentally collecting of PII data
 
-        <CheckboxField field="auth_needed_to_vote" label="Require Auth to Vote">
-          Participants cannot vote without first connecting either Facebook or
-          Twitter
-        </CheckboxField>
+        // changed also in polis-server constants.ts: auth_needed_to_write = false
+        // <CheckboxField
+        //   field="auth_needed_to_write"
+        //   label="Require Auth to Comment">
+        //   Participants cannot submit comments without first connecting either
+        //   Facebook or Twitter
+        // </CheckboxField>
+
+        // changed also in polis-server constants.ts: auth_needed_to_vote = false
+        // <CheckboxField field="auth_needed_to_vote" label="Require Auth to Vote">
+        //   Participants cannot vote without first connecting either Facebook or
+        //   Twitter
+        // </CheckboxField>
+        }
       </Box>
     )
   }
