--- ./client-admin/src/components/landers/lander-footer.js
+++ ../patches/client-admin/src/components/landers/lander-footer.js
@@ -18,8 +18,14 @@
           {emoji('🇺🇸')}, with contributions from around the {emoji('🌍🌏🌎')}
         </Box>
         <Box sx={{ mb: [2, null, 3] }}>
-          © {new Date().getFullYear()} The Authors <Link href="tos">TOS</Link>{' '}
-          <Link href="privacy">Privacy</Link>
+          © {new Date().getFullYear()} The Authors 
+          {' '}
+
+          { // remove TOS and replace privacy link with external DigiFinland site privary policy page
+          // <Link href="tos">TOS</Link>                    
+          // <Link href="privacy">Privacy</Link>
+          }
+          <Link target="_blank" href="https://digifinland.fi/tietosuoja/">Privacy</Link>
         </Box>
       </Box>
     )
