--- ./client-admin/src/components/landers/password-reset-init.js
+++ ../patches/client-admin/src/components/landers/password-reset-init.js
@@ -7,8 +7,10 @@
 
 import StaticLayout from './lander-layout'
 
+import { withTranslation } from 'react-i18next';
+
 @connect()
-class PasswordResetInit extends React.Component {
+class PasswordResetInitTranslated extends React.Component {
   handleClick(e) {
     e.preventDefault()
 
@@ -20,17 +22,18 @@
   }
 
   render() {
+    const { t } = this.props;
     return (
       <StaticLayout>
-        <h1> Password Reset</h1>
-        <form>
+        <h1>{t('password.title')}</h1>
+        <form style={{marginBottom: "50px"}}>
           <input
             ref={(c) => (this.email = c)}
-            placeholder="email"
+            placeholder={t('password.email')}
             type="text"
           />
           <button onClick={this.handleClick.bind(this)}>
-            Send password reset email
+            {t('password.send_email')}
           </button>
         </form>
       </StaticLayout>
@@ -38,8 +41,11 @@
   }
 }
 
-PasswordResetInit.propTypes = {
-  dispatch: PropTypes.func
+PasswordResetInitTranslated.propTypes = {
+  dispatch: PropTypes.func,
+  t: PropTypes.func.isRequired
 }
 
-export default PasswordResetInit
+// export default PasswordResetInit
+const PasswordResetInit = withTranslation()(PasswordResetInitTranslated);
+export default withTranslation()(PasswordResetInit);
