--- ./client-admin/src/components/landers/password-reset.js
+++ ../patches/client-admin/src/components/landers/password-reset.js
@@ -6,8 +6,10 @@
 import { doPasswordReset } from '../../actions'
 import StaticLayout from './lander-layout'
 
+import { withTranslation } from 'react-i18next';
+
 @connect()
-class PasswordReset extends React.Component {
+class PasswordResetTranslated extends React.Component {
   handleClick(e) {
     e.preventDefault()
     const attrs = {
@@ -24,22 +26,23 @@
   }
 
   render() {
+    const { t } = this.props;
     return (
       <StaticLayout>
-        <h1>Password Reset</h1>
-        <form>
+        <h1>{t('password.title')}</h1>
+        <form style={{marginBottom: "50px"}}>
           <input
             ref={(c) => (this.password = c)}
-            placeholder="new password"
+            placeholder={t('password.new')}
             type="password"
           />
           <input
             ref={(c) => (this.passwordRepeat = c)}
-            placeholder="repeat new password"
+            placeholder={t('password.new_repeat')}
             type="password"
           />
           <button onClick={this.handleClick.bind(this)}>
-            Set new password
+            {t('password.new_set')}
           </button>
         </form>
       </StaticLayout>
@@ -47,9 +50,12 @@
   }
 }
 
-PasswordReset.propTypes = {
+PasswordResetTranslated.propTypes = {
   dispatch: PropTypes.func,
-  location: PropTypes.object
+  location: PropTypes.object,
+  t: PropTypes.func.isRequired
 }
 
-export default PasswordReset
+// export default PasswordReset
+const PasswordReset = withTranslation()(PasswordResetTranslated);
+export default withTranslation()(PasswordReset);
