--- ./client-admin/src/components/landers/signout.js
+++ ../patches/client-admin/src/components/landers/signout.js
@@ -11,7 +11,7 @@
 @connect((state) => state.signout)
 class SignOut extends React.Component {
   componentDidMount() {
-    this.props.dispatch(doSignout('/home'))
+    this.props.dispatch(doSignout('/signin')) // DF customization, due to WAF restrictions redirect back to /signin instead of /home
   }
 
   render() {
