--- ./client-participation/js/strings.js
+++ ../patches/client-participation/js/strings.js
@@ -4,6 +4,10 @@
 var Utils = require("./util/utils");
 
 var translations = {
+  // Finnish
+  fi: require("./strings/fi.js"),
+  // Swedish
+  sv: require("./strings/sv.js"),
   // Arabic
   ar: require("./strings/ar.js"),
   // Welsh
@@ -79,7 +83,16 @@
   prioritized.forEach(function (languageCode) {
     if (languageCode.match(/^en/)) {
       _.extend(strings, translations.en_us);
-    } else if (languageCode.match(/^ja/)) {
+    } else if (
+      languageCode.match(/^fi/) ||
+      languageCode.match(/^fi-FI/)) {
+      _.extend(strings, translations.fi);
+    } else if (
+      languageCode.match(/^sv/) ||
+      languageCode.match(/^sv-SE/)) {
+      _.extend(strings, translations.sv);
+    }
+    else if (languageCode.match(/^ja/)) {
       _.extend(strings, translations.ja);
     } else if (
       languageCode.match(/^zh-CN/) ||
