--- ./client-participation/js/strings.js
+++ ../patches/client-participation/js/strings.js
@@ -63,6 +63,13 @@
 
   // Romanian & Moldovan
   ro: require("./strings/ro.js"),
+
+  // Finnish
+  fi: require("./strings/fi.js"),
+
+  // Swedish
+  sv: require("./strings/sv.js"),
+
 };
 
 
@@ -86,6 +93,16 @@
     if (languageCode.match(/^en/)) {
       _.extend(strings, translations.en_us);
     }
+    else if (
+      languageCode.match(/^fi/) ||
+      languageCode.match(/^fi-FI/)) {
+      _.extend(strings, translations.fi);
+    }
+    else if (
+      languageCode.match(/^sv/) ||
+      languageCode.match(/^sv-SE/)) {
+      _.extend(strings, translations.sv);
+    }
     else if (languageCode.match(/^ja/)) {
       _.extend(strings, translations.ja);
     }
