--- ./client-admin/src/components/interior-header.js
+++ ../patches/client-admin/src/components/interior-header.js
@@ -4,8 +4,16 @@
 import { Link } from 'react-router-dom'
 import Logomark from './framework/logomark'
 
-class InteriorHeader extends React.Component {
+import { withTranslation } from 'react-i18next';
+
+class InteriorHeaderTranslated extends React.Component {
   render() {
+    const { t, i18n } = this.props;
+    document.documentElement.lang = i18n.language; //This is too slow for google translate
+    const changeLanguage = (lng) => {
+        i18n.changeLanguage(lng);
+        document.documentElement.lang = lng;
+    };
     return (
       <Box>
         <Box
@@ -27,14 +35,24 @@
             />
             Polis
           </Link>
+          
           <Link id="signoutLink" sx={{ variant: 'links.header' }} to="/signout">
-            sign out
+            {t('header.signout')}
           </Link>
+          
         </Box>
+        <div style={{display:"flex", justifyContent:"flex-end"}}>
+            <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('fi')}>FI</button>
+            <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('sv')}>SV</button>
+            <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('en')}>EN</button>
+          </div>
         <Box>{this.props.children}</Box>
       </Box>
     )
   }
 }
 
-export default InteriorHeader
+//export default InteriorHeader
+const InteriorHeader = withTranslation()(InteriorHeaderTranslated);
+export default withTranslation()(InteriorHeader);
+
