--- ./client-admin/src/index.js
+++ ../patches/client-admin/src/index.js
@@ -2,7 +2,7 @@
 
 import $ from 'jquery'
 
-import React from 'react'
+import React, { Suspense } from 'react';
 import ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
 
@@ -13,11 +13,15 @@
 
 import { BrowserRouter as Router, Route } from 'react-router-dom'
 
+// import i18n (needs to be bundled ;)) 
+import './i18n';
+
 const store = configureStore()
 
 class Root extends React.Component {
   render() {
     return (
+    <Suspense fallback="loading">
       <ThemeProvider theme={theme}>
         <Provider store={store}>
           <Router>
@@ -25,6 +29,7 @@
           </Router>
         </Provider>
       </ThemeProvider>
+      </Suspense>
     )
   }
 }
