--- ./client-admin/src/components/conversation-admin/comment-moderation/comment.js
+++ ../patches/client-admin/src/components/conversation-admin/comment-moderation/comment.js
@@ -5,12 +5,14 @@
 import { connect } from 'react-redux'
 import { Flex, Box, Text, Button, Card, Link } from 'theme-ui'
 
+import { withTranslation } from 'react-i18next';
+
 @connect((state) => {
   return {
     conversation: state.zid_metadata.zid_metadata
   }
 })
-class Comment extends React.Component {
+class CommentTranslated extends React.Component {
   onAcceptClicked() {
     this.props.acceptClickHandler(this.props.comment)
   }
@@ -24,6 +26,7 @@
   }
 
   render() {
+    const { t } = this.props;
     return (
       <Card sx={{ mb: [3], minWidth: '35em' }}>
         <Box>
@@ -53,12 +56,12 @@
                 target="_blank"
                 sx={{ mr: [2] }}
                 href="https://compdemocracy.org/metadata">
-                {this.props.isMetaCheckbox ? 'metadata' : null}
+                {this.props.isMetaCheckbox ? t('moderate.metadata') : null}
               </Link>
               {this.props.isMetaCheckbox ? (
                 <input
                   type="checkbox"
-                  label="metadata"
+                  label={t('moderate.metadata')}
                   ref={(c) => (this.is_meta = c)}
                   checked={this.props.comment.is_meta}
                   onChange={this.onIsMetaClicked.bind(this)}
@@ -72,7 +75,7 @@
   }
 }
 
-Comment.propTypes = {
+CommentTranslated.propTypes = {
   dispatch: PropTypes.func,
   acceptClickHandler: PropTypes.func,
   rejectClickHandler: PropTypes.func,
@@ -82,10 +85,13 @@
   rejectButton: PropTypes.bool,
   rejectButtonText: PropTypes.string,
   isMetaCheckbox: PropTypes.bool,
+  t: PropTypes.func.isRequired,
   comment: PropTypes.shape({
     txt: PropTypes.string,
     is_meta: PropTypes.bool
   })
 }
 
-export default Comment
+// export default Comment
+const Comment = withTranslation()(CommentTranslated);
+export default withTranslation()(Comment);
\ No newline at end of file
