--- ./client-report/src/components/lists/commentList.jsx
+++ ../patches/client-report/src/components/lists/commentList.jsx
@@ -3,6 +3,8 @@
 import React from "react";
 import * as globals from "../globals";
 
+import { useTranslation } from 'react-i18next';
+
 const BarChartCompact = ({ comment, voteCounts, nMembers, voteColors }) => {
   if (!comment) return null;
 
@@ -31,21 +33,23 @@
   const disagreeSaw = (disagrees / sawTheComment) * w;
   const passSaw = (passes / sawTheComment) * w;
 
-  const agreeString = (agreeSaw << 0) + "%";
-  const disagreeString = (disagreeSaw << 0) + "%";
-  const passString = (passSaw << 0) + "%";
+  const agreeString = (agreeSaw << 0) + "% ";
+  const disagreeString = (disagreeSaw << 0) + "% ";
+  const passString = (passSaw << 0) + "% ";
 
+  const { t } = useTranslation();
+  
   return (
     <div
       title={
         agreeString +
-        " Agreed\n" +
+        t('comment_list.agreed')+"\n" +
         disagreeString +
-        " Disagreed\n" +
+        t('comment_list.disagreed')+"\n" +
         passString +
-        " Passed\n" +
+        t('comment_list.passed')+"\n" +
         sawTheComment +
-        " Respondents"
+        t('comment_list.respondents')
       }
     >
       <svg width={101} height={10} style={{ marginRight: 30 }}>
@@ -153,6 +157,8 @@
 
 const CommentList = ({ comments, math, ptptCount, tidsToRender, voteColors }) => {
 
+  const { t } = useTranslation();
+  
   const getGroupLabels = () => {
     function makeLabel(key, label, numMembers) {
       return (
@@ -181,7 +187,7 @@
     let labels = [];
 
     // totals
-    labels.push(makeLabel(99, "Overall", ptptCount));
+    labels.push(makeLabel(99, t('comment_list.overall'), ptptCount));
 
     Object.entries(math["group-votes"]).forEach(([key, g]) => {
       const i = parseInt(key, 10);
@@ -216,7 +222,7 @@
             textTransform: "uppercase",
           }}
         >
-          Statement
+          {t('comment_list.statement')}
         </span>
 
         {getGroupLabels()}
