--- ./client-report/src/components/lists/commentList.js
+++ ../patches/client-report/src/components/lists/commentList.js
@@ -4,6 +4,8 @@
 import _ from "lodash";
 import * as globals from "../globals";
 
+import { useTranslation, withTranslation } from 'react-i18next';
+
 const BarChartCompact = ({ comment, voteCounts, nMembers, voteColors }) => {
   if (!comment) return null;
 
@@ -32,21 +34,19 @@
   const disagreeSaw = (disagrees / sawTheComment) * w;
   const passSaw = (passes / sawTheComment) * w;
 
-  const agreeString = (agreeSaw << 0) + "%";
-  const disagreeString = (disagreeSaw << 0) + "%";
-  const passString = (passSaw << 0) + "%";
+  const agreeString = (agreeSaw << 0) + "% ";
+  const disagreeString = (disagreeSaw << 0) + "% ";
+  const passString = (passSaw << 0) + "% ";
+
+  const { t } = useTranslation();
 
   return (
     <div
       title={
-        agreeString +
-        " Agreed\n" +
-        disagreeString +
-        " Disagreed\n" +
-        passString +
-        " Passed\n" +
-        sawTheComment +
-        " Respondents"
+        agreeString + t('comment_list.agreed')+"\n"+
+        disagreeString + t('comment_list.disagreed')+"\n"+
+        passString + t('comment_list.passed')+"\n"+
+        sawTheComment + t('comment_list.respondents')
       }
     >
       <svg width={101} height={10} style={{ marginRight: 30 }}>
@@ -159,7 +159,7 @@
   );
 };
 
-class CommentList extends React.Component {
+class CommentListTranslated extends React.Component {
   getGroupLabels() {
     function makeLabel(key, label, numMembers) {
       return (
@@ -186,9 +186,11 @@
       );
     }
     let labels = [];
+    
+    const { t } = this.props;
 
     // totals
-    labels.push(makeLabel(99, "Overall", this.props.ptptCount));
+    labels.push(makeLabel(99, t('comment_list.overall'), this.props.ptptCount));
 
     _.each(this.props.math["group-votes"], (g, i) => {
       labels.push(makeLabel(i, globals.groupLabels[i], g["n-members"]));
@@ -199,7 +201,7 @@
 
   render() {
     const comments = _.keyBy(this.props.comments, "tid");
-
+    const { t } = this.props;
     return (
       <div>
         <div
@@ -220,7 +222,7 @@
               textTransform: "uppercase",
             }}
           >
-            Statement
+            {t('comment_list.statement')}
           </span>
 
           {this.getGroupLabels()}
@@ -241,4 +243,6 @@
   }
 }
 
-export default CommentList;
+//export default CommentList;
+const CommentList = withTranslation()(CommentListTranslated);
+export default withTranslation()(CommentList);
