--- ./client-report/src/components/lists/participantGroup.jsx
+++ ../patches/client-report/src/components/lists/participantGroup.jsx
@@ -4,6 +4,8 @@
 import * as globals from "../globals.js";
 import CommentList from "./commentList.jsx";
 
+import { useTranslation } from 'react-i18next';
+
 const ParticipantGroup = ({
   gid,
   groupComments,
@@ -16,10 +18,10 @@
   math,
   voteColors,
 }) => {
-
+  const { t } = useTranslation();
   let groupLabel = groupName;
   if (typeof groupLabel === "undefined") {
-    groupLabel = "Group " + globals.groupLabels[gid];
+    groupLabel = t('participant_groups.group')+' '+ globals.groupLabels[gid];
   }
 
   return (
@@ -28,9 +30,9 @@
         width: "100%",
       }}>
       <p style={globals.secondaryHeading}>
-        {groupLabel}: {groupVotesForThisGroup["n-members"]} participants
+        {groupLabel}: {groupVotesForThisGroup["n-members"]} {t('participant_groups.participants')}
       </p>
-      <p style={globals.paragraph}> Statements which make this group unique, by their votes: </p>
+      <p style={globals.paragraph}>{t('participant_groups.grp_desc')}</p>
       <CommentList
         conversation={conversation}
         ptptCount={ptptCount}
