--- ./client-report/src/components/lists/uncertainty.js	2023-11-10 18:35:57.628264848 +0200
+++ ../patches/client-report/src/components/lists/uncertainty.js	2023-11-10 18:34:12.884579400 +0200
@@ -4,22 +4,19 @@
 import CommentList from "./commentList";
 import * as globals from "../globals";
 // import style from "../../util/style";
+import { useTranslation } from 'react-i18next';
 
 const Uncertainty = ({conversation, comments, ptptCount, uncertainty, formatTid, math, voteColors}) => {
 
   if (!conversation) {
     return <div>Loading Uncertainty...</div>
   }
+  const { t } = useTranslation();
   return (
     <div>
-      <p style={globals.primaryHeading}> Areas of uncertainty </p>
-      <p style={globals.paragraph}>
-        Across all {ptptCount} participants, there was uncertainty about the following statements.
-        Greater than 30% of participants who saw these statements &apos;passed&apos;.
-      </p>
-      <p style={globals.paragraph}>
-        Areas of uncertainty can provide avenues to educate and open dialogue with your community.
-      </p>
+      <p style={globals.primaryHeading}>{t('uncertainty.title')}</p>
+      <p style={globals.paragraph}>{t('uncertainty.text', { param1: ptptCount})}</p>
+      <p style={globals.paragraph}>{t('uncertainty.desc')}</p>
       <div style={{marginTop: 50}}>
         <CommentList
           conversation={conversation}
