--- ./client-report/src/components/controls/controls.jsx
+++ ../patches/client-report/src/components/controls/controls.jsx
@@ -1,8 +1,10 @@
 // Copyright (C) 2012-present, The Authors. This program is free software: you can redistribute it and/or  modify it under the terms of the GNU Affero General Public License, version 3, as published by the Free Software Foundation. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details. You should have received a copy of the GNU Affero General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+
 import Checkbox from "../framework/checkbox.jsx";
 import React from "react";
 import settings from "../../settings";
+import { useTranslation, i18n } from 'react-i18next';
 
 const Controls = ({
   onAutoRefreshEnabled,
@@ -19,11 +21,24 @@
       onAutoRefreshDisabled();
     }
   };
+  
+  const { t } = useTranslation();
+  
+  if (i18n && i18n.language) {
+    document.documentElement.lang = i18n.language; //This is too slow for google translate
+  }
+  const changeLanguage = (lng) => {
+    i18n.changeLanguage(lng);
+    document.documentElement.lang = lng;
+  };
 
   return (
     <div>
+      <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('fi')}>FI</button>
+      <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('sv')}>SV</button>
+      <button style={{margin: "10px"}} type="button" onClick={() => changeLanguage('en')}>EN</button>
       <Checkbox
-        label="auto-refresh"
+        label={t('controls.auto_refresh')}
         disabled={false}
         checked={autoRefreshEnabled}
         clickHandler={checkboxGroupChanged}
@@ -32,7 +47,7 @@
         color={settings.polisBlue}
       />
       <Checkbox
-        label="color blind mode"
+        label={t('controls.color_blind_mode')}
         disabled={false}
         checked={colorBlindMode}
         clickHandler={handleColorblindModeClick}
